### PR TITLE
[GEMFILE] ⚡️ support local_dev ⚡️

### PR DESCRIPTION
**Pull request by hjanuschka**
whenever i touch any of the sub-gems - i need to add the gem manually to the Gemfile.
and evertime on some git operations, i have to stash those changes.

with this change you can do

```
FASTLANE_LOCAL_DEV=1 bundle update
FASTLANE_LOCAL_DEV=1 bundle install 
FASTLANE_LOCAL_DEV=1 bundle exec fastlane demo
```

it will use all gems from the local folders -  for sure you have to call the `update` `install` with the env too.

let me know if i am totally on a wrong track - and if there is an easier way.

edit: will add docs, once i know if it is ok to go with this PR

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,3 +3,15 @@ source "https://rubygems.org"
 gemspec path: "fastlane"
 
 gem "danger", "~> 0.10"
+
+if ENV["FASTLANE_LOCAL_DEV"]
+  require "fastlane/tools.rb"
+  local_deps = Fastlane::TOOLS + [:fastlane_core, :credentials_manager]
+  local_deps.each do |dep|
+    if dep == :fastlane
```
Comment (KrauseFx) [inline Gemfile]: This should be
```ruby
next if dep == :fastlane
```
for more 💎

**Review by KrauseFx — CHANGES_REQUESTED** [review]

**Comment by KrauseFx** [comment]
Great idea 👍

**Comment by hjanuschka** [comment]
@KrauseFx  Done.

**Review by KrauseFx — APPROVED** [review]

**Comment by KrauseFx** [comment]
This looks great, thanks 👍

**Pull request merged**